src/Chapter3.hs

```diff
@@ -480,6 +509,50 @@ After defining the city, implement the following functions:
    and at least 10 living __people__ inside in all houses of the city in total.
 -}
 
+data Castle
+  = None
+  | Castle {castleName :: String}
+  | CastleWithWalls {castleName :: String}
+  deriving (Show)
```

Comment:
Records inside sum types are considered an anti-pattern. Because, in your case, the `castleName` function will be partial. When defining a record like this one, the compiler provides a function `castleName :: Castle -> String`. But what `String` can it return for the `None` constructor? That's why this function is partial, and usage of such partial record fields is generally discouraged.